### ISSUE TITLE
automatically build Docker images

### ISSUE THREAD

**Issue by anarcat** [whipper-team/whipper#301]
A key thing of Docker is that there is the [Docker hub](https://hub.docker.com/) which automatically builds Docker images for you. This allows users to use your software without having to download (or build!) the source.

There are already two "whipper" Docker images when you search on the hub [whipper7/test](https://hub.docker.com/r/whipper7/test/) (probably unrelated) and [dartagnan/whipper](https://hub.docker.com/r/dartagan/whipper/) unofficial build. This means that Docker users are likely to use an unofficial (or totally unrelated) image instead of the one you provide here.

Enabling automated builds of your image is fairly simple:

 1. sign in to [hub.docker.com](https://hub.docker.com/) (creating a new account first if not already done)
 2. in the top-right menu, click on `Create` then `Create automated build`
 3. this will ask you to link your GitHub account if not already done. do so. you can also grant other GitHub members that have access to your repo later as well so others can operate the build
 4. Once the account is linked, you can `Create Auto-build Githb` there
 5. then you pick this repository in the list
 6. fill in the details as necessary (optional) and click `Save`

Then the build should be automatically triggered when changes are performed in the GitHub repository.

I can assist with the creation process if you grant me access, but it would of course be better if you would get familiar with it.

I went through the above process and created the [anarcat/whipper](https://hub.docker.com/r/anarcat/whipper/) repository based on my fork, as an example. I will delete it as soon as this issue is completed.

Thank you for your attention and hard work!

**Comment by anarcat** [comment]
once that is done, the Docker instructions in the README would become:

```
docker pull whipper/whipper
```

instead of the `docker build` command.

**Comment by JoeLametta** [comment]
This is an interesting suggestion, thanks!
I don't know: maybe it would be a good idea to create a Docker Hub account just for whipper?

**Comment by anarcat** [comment]
On 2018-10-03 08:54:37, JoeLametta wrote:
> This is an interesting suggestion, thanks!
> I don't know: maybe it would be a good idea to create a Docker Hub account just for whipper?

I'd say it'd be a good idea to create a org on github, and then each
maintainer can have an individual account on hub for whipper...

but yes, what I have done in other projects is create an org account on
Docker Hub for the team.

ideally, there'd be a 1-to-1 mapping between the two.

**Comment by JoeLametta** [comment]
@anarcat Sorry, I've got no experience with Docker... :rofl: 
Let me know if I got it right: are you suggesting to move whipper's repository ownership to a GitHub organization, sign up for a personal Docker Hub account then create a Docker Hub organization (for whipper)?

**Comment by anarcat** [comment]
On 2018-10-06 03:14:01, JoeLametta wrote:
> @anarcat Sorry, I've got no experience with Docker... :rofl: 
> Let me know if I got it right: are you suggesting to [...]

> 1. move whipper's repository ownership to a GitHub organization

optionally, yes.

> 2. sign up for a personal Docker Hub account

definitely, yes.

> 3. then create a Docker Hub organization (for whipper)?

ideally, yes.

the core idea, in any case, is to create an automated build. this can be
done with only step 2 above.

**Comment by JoeLametta** [comment]
All right, in case I perform step 1 later on (after having already enabled the automated build for the current repository, step 2 & 3 done), I expect I can simply switch the Docker Hub target to the GitHub org. repo without having to fiddle with the accounts.

**Comment by anarcat** [comment]
> All right, in case I perform step 1 later on (after having already enabled the automated build for the current repository, step 2 & 3 done), I expect I can simply switch the Docker Hub target to the GitHub org. repo without having to fiddle with the accounts.

well, you'd only two step 2: create a personal docker hub account and hook the build there. when you setup a new org on GitHub, you'd just setup a new, matching, org on docker hub as well and ditch the old one.

**Comment by anarcat** [comment]
On 2018-10-16 16:25:04, JoeLametta wrote:
> Fine, so I sign up on Docker Hub for a personal account, then create an organization (using the account just created) and link the whipper's GitHub (org) repo on to it, right? (sorry, I've never used Docker/Docker Hub before)

that sounds about right! :)

just try it out and let me know what fails...

**Comment by JoeLametta** [comment]
@anarcat [joelametta/whipper](https://hub.docker.com/r/joelametta/whipper/): done?

:wink:

**Comment by anarcat** [comment]
that looks awesome! congrats :)

**Comment by JoeLametta** [comment]
> that looks awesome! congrats :)

I've just updated the README: special thanks in https://github.com/whipper-team/whipper/commit/f9cb1262705acbe67e53882a6c8a9c75e022953e 's description. :tada:
